production/kubernetes/agent-bare.yaml

```diff
@@ -22,26 +22,22 @@ spec:
         name: grafana-pdc-agent
     spec:
       containers:
-      - args:
-        - -i
-        - /var/run/secrets/grafana-pdc-agent/key
-        - ${SLUG}@${PDC_GATEWAY}
-        - -p
-        - "22"
-        - -o
-        - BatchMode=yes
-        - -o
-        - UserKnownHostsFile=/var/run/secrets/grafana-pdc-agent/known_hosts
-        - -o
-        - CertificateFile=/var/run/secrets/grafana-pdc-agent/cert.pub
-        - -o
-        - ServerAliveInterval=15
-        - -R
-        - "0"
-        - -vvv
+      - name: grafana-pdc-agent
+        env:
+        - name: TOKEN
+          valueFrom:
+            secretKeyRef:
+              key: token
+              name: grafana-pdc-agent
```

Comment:
🤔 I am tempted to put all the variable in the secret, and remove any templating from the deployment. This would remove the need for the bash script:

```
kubectl create secret generic ...
kubectl apply -f https://raw.githubusercontent.com/grafana/pdc-agent/main/production/kubernetes/agent-secret.yaml
```

Thoughts ?

Reply:
I would love to get rid of the bash scripts. Lets do it